be/src/storage/txn_manager.cpp

```diff
@@ -207,116 +201,54 @@ Status TxnManager::commit_txn(KVStore* meta, TPartitionId partition_id, TTransac
-                // case 1: user commit rowset, then the load id must be equal
-                TabletTxnInfo& load_info = load_itr->second;
-                rowset_ptr = load_info.rowset;
+        std::lock_guard txn_lock(_get_txn_lock(transaction_id));
+        if (tablet->updates() != nullptr) {
+            StarRocksMetrics::instance()->update_rowset_commit_request_total.increment(1);
+            auto st = tablet->rowset_commit(version, rowset);
+            if (!st.ok()) {
+                StarRocksMetrics::instance()->update_rowset_commit_request_failed.increment(1);
+                return st;
             }
-        }
-    }
-    // save meta need access disk, it maybe very slow, so that it is not in global txn lock
-    // it is under a single txn lock
-    if (rowset_ptr != nullptr) {
-        // TODO(ygl): rowset is already set version here, memory is changed, if save failed
-        // it maybe a fatal error
-        rowset_ptr->make_visible(version);
-        auto& rowset_meta_pb = rowset_ptr->rowset_meta()->get_meta_pb();
-        Status st = RowsetMetaManager::save(meta, tablet_uid, rowset_meta_pb);
-        if (!st.ok()) {
-            LOG(WARNING) << "Fail to save committed rowset. "
-                         << "tablet_id: " << tablet_id << ", txn_id: " << transaction_id
-                         << ", rowset_id: " << rowset_ptr->rowset_id();
-            return Status::InternalError(
-                    fmt::format("Fail to save committed rowset. tablet_id: {}, txn_id: {}", tablet_id, key.second));
-        }
-    } else {
-        return Status::NotFound(fmt::format("Not found txn. tablet_id: {}, txn_id:{} ", tablet_id, transaction_id));
-    }
-    {
-        std::unique_lock wrlock(_get_txn_map_lock(transaction_id));
-        txn_tablet_map_t& txn_tablet_map = _get_txn_tablet_map(transaction_id);
-        auto it = txn_tablet_map.find(key);
-        if (it != txn_tablet_map.end()) {
-            it->second.erase(tablet_info);
-            VLOG(1) << "publish txn successfully."
-                    << " partition_id: " << key.first << ", txn_id: " << key.second
-                    << ", tablet: " << tablet_info.to_string() << ", rowsetid: " << rowset_ptr->rowset_id()
-                    << ", version: " << version.first << "," << version.second;
-            if (it->second.empty()) {
-                txn_tablet_map.erase(it);
-                _clear_txn_partition_map_unlocked(transaction_id, partition_id);
+        } else {
+            // TODO(ygl): rowset is already set version here, memory is changed, if save failed
+            // it maybe a fatal error
+            rowset->make_visible({version, version});
+            auto& rowset_meta_pb = rowset->rowset_meta()->get_meta_pb();
+            Status st = RowsetMetaManager::save(tablet->data_dir()->get_meta(), tablet->tablet_uid(), rowset_meta_pb);
+            if (!st.ok()) {
+                LOG(WARNING) << "Fail to save committed rowset. "
```

Comment:
It is best not to output logs within the lock scope

Reply:
This part is the same code as before, what do you propose instead? Just remove this log is an option